accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,245 @@
+        environment VARCHAR(128) NULL
+                      CONSTRAINT suse_srvcocoatt_rep_env_ck
+                        CHECK(environment IN ('kvm_amd_epyc_milan', 'kvm_amd_epyc_genoa')),
+        status      VARCHAR(32) NOT NULL
+                      CONSTRAINT suse_srvcocoatt_rep_st_ck
+                        CHECK(status IN ('pending', 'succeeded', 'failed')),
+        data        JSONB NOT NULL, -- machine readable data
+        created     TIMESTAMPTZ DEFAULT (current_timestamp) NOT NULL,
+        modified    TIMESTAMPTZ DEFAULT (current_timestamp) NOT NULL
+);
+
+CREATE SEQUENCE suse_srvcocoatt_rep_id_seq;
+
+CREATE INDEX suse_srvcocoatt_rep_sid_idx
+  ON suseServerCoCoAttestationReport (server_id);
+
+CREATE INDEX suse_srvcocoatt_rep_stenv_idx
+  ON suseServerCoCoAttestationReport (status, environment);
+```
+
+The result table has the result for every check performed on the report.
+This is a many-to-one relation as many checks can be performed for one report.
+Every result has a type, status and summary which can be shown in a user interface.
+Optional it can have details if applicable.
+
+
+```sql
+CREATE TABLE suseServerCoCoAttestationResult
+(
+        id          NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_res_id_pk PRIMARY KEY,
+        report_id   NUMERIC NOT NULL
+                      CONSTRAINT suse_srvcocoatt_res_rid_fk
+                        REFERENCES suseServerCoCoAttestationReport (id),
+        result_type VARCHAR(32) NOT NULL
+                      CONSTRAINT suse_srvcocoatt_res_ty_ck
+                        CHECK(result_type IN ('sev_snp', 'secure_boot')),
+        status      VARCHAR(32) NOT NULL
+                      CONSTRAINT suse_srvcocoatt_res_st_ck
+                        CHECK(status IN ('succeeded', 'failed')),
+        summary     VARCHAR(256) NOT NULL,
+        details     TEXT NULL,
+        attested    TIMESTAMPTZ NULL
+);
+
+CREATE SEQUENCE use_srvcocoatt_res_id_seq;
+
+CREATE INDEX use_srvcocoatt_res_rid_idx
+  ON suseServerCoCoAttestationResult (report_id);
+```
+
+
+## SNP Attestation of KVM instances
+
+This attestation is using the tool `snpguest` which is provided in SUSE Linux Enterprise Server 15 SP6 and later.
+It is made for AMD EPYC CPUs of type Milan and Genoa.
+The host and the hypervisor must support Confidential VMs to perform a successful attestation.
+
+The tool and the possible attestation workflows are described in the [README](https://github.com/virtee/snpguest/blob/main/README.md).
+
+The proposed workflow for attestation using SUSE Manager is:
+
+![SNP Guest Attestation](images/coco-attestation/CoCoAttestation.png)
+
+- Apply a salt state on the VM
+  - send a nonce file
+  - request the attestation report
+- Store the data into the database
+  - Status: pending
+  - Store nonce and report data
+  - assign to a systemid and an execution date (create date of the report)
+- Run a Container which does the attestation
```

Comment:
It was planned that a container should do all the check for 1 environment. 
But if 2 checks in one environment require quite different software it could make sense to have containers per check which needs to be performed.
But that would require that the java code knows about which environment requires which check.
Or we define this in a database table. I will think about this solution.

Reply:
I will change the RFC to follow this new way.